File: nagasena/src/org/openexi/proc/io/compression/ScannerChannel.java
package org.openexi.proc.io.compression;

import java.util.ArrayList;

import org.openexi.proc.common.Channel;

class ScannerChannel extends Channel {
  
  final ArrayList<EXIEventValueReference> values;

  ScannerChannel(int firstPos, int blockNum) {
    super(firstPos, blockNum);
    values = new ArrayList<EXIEventValueReference>();
  }

  @Override
  public void reset(int firstPos, int blockNum) {
    super.reset(firstPos, blockNum);
    values.clear();
  }

}

File: nagasenacs/src/Nagasena/Proc/IO/Compression/ScannerChannel.cs
using System.Collections.Generic;

using Channel = Nagasena.Proc.Common.Channel;

namespace Nagasena.Proc.IO.Compression {

  internal class ScannerChannel : Channel {

    internal readonly List<EXIEventValueReference> values;

    internal ScannerChannel(int firstPos, int blockNum) : base(firstPos, blockNum) {
      values = new List<EXIEventValueReference>();
    }

    public override void reset(int firstPos, int blockNum) {
      base.reset(firstPos, blockNum);
      values.Clear();
    }

  }

}